lib/device-connections-factory.js

```diff
@@ -37,29 +37,57 @@ class iProxy {
       }
     });
     const status = new B((resolve, reject) => {
-      this.serverSocket.once('listening', resolve);
-      this.serverSocket.once('error', reject);
+      this.localServer.once('listening', resolve);
+      this.localServer.once('error', reject);
     });
-    this.serverSocket.listen(this.localport);
-    await status;
-    this.serverSocket.on('error', (e) => this.log.warn(e.message));
+    this.localServer.listen(this.localport);
+    try {
+      await status;
+    } catch (e) {
+      this.localServer = null;
+      throw e;
+    }
+    this.localServer.on('error', (e) => this.log.warn(e.message));
+
+    this.onBeforeProcessExit = () => {
+      if (this.localServer?.listening) {
+        this.localServer.close((e) => {
+          if (e) {
+            this.log.debug(e);
+          } else {
+            this.log.info(`The connection to ${this.udid} has been closed`);
+          }
+          this.localServer = null;
+        });
+      }
+    };
+    // Make sure we free up the socket on process exit
+    process.on('beforeExit', this.onBeforeProcessExit);
```

Comment:
Need to `bind()` here or `this` will be wrong.

Reply:
I would say this is fine, because onBeforeProcessExit is a lambda rather than a normal function